File: fix_herbs.py
#!/usr/bin/env python3
import re

# ============ FIX HERBS ============
with open('/home/justin/.openclaw/workspace/tcm_ai/vercel_tcm_frontend/app/[locale]/herbs/page.tsx', 'r') as f:
    c = f.read()

# Add herbView state
c = c.replace(
    "const [selectedCat, setSelectedCat] = useState('')",
    "const [selectedCat, setSelectedCat] = useState('')\n  const [herbView, setHerbView] = useState<'home' | 'list'>('home')"
)

# Find the category pills section and wrap with conditional
old_cat_pills = '''        {/* Category pills */}
        <div style={{ padding: '10px 14px 0', overflowX: 'auto', WebkitOverflowScrolling: 'touch' }}>
          <div style={{ display: 'flex', gap: 6, minWidth: 'max-content' }}>
            {CATEGORIES.map(cat => (
              <button key={cat.key} onClick={() => setSelectedCat(cat.key)} style={{
                padding: '6px 12px', borderRadius: 20, border: 'none', cursor: 'pointer', fontSize: 12, fontWeight: 700,
                backgroundColor: selectedCat === cat.key ? '#FFFEF9' : 'rgba(255,254,249,0.15)',
                color: selectedCat === cat.key ? '#1a3A2C' : 'rgba(255,254,249,0.85)',
                whiteSpace: 'nowrap', flexShrink: 0,
              }}>
                {cat.emoji} {cat.label}
              </button>
            ))}
          </div>
        </div>

        {/* Count */}
        <div style={{ padding: '8px 16px 0', fontSize: 12, color: 'rgba(255,254,249,0.65)' }}>
          {loading ? '載入中...' : `${filtered.length} 味中藥`}
        </div>
      </div>

      {/* Content */}
      <div style={{ padding: '12px 14px 100px' }}>
        {loading ? ('''

new_cat_pills = '''        {herbView === 'home' && (
          <div style={{ padding: '16px 14px 100px' }}>
            <div style={{ fontSize: 13, color: 'rgba(255,254,249,0.65)', marginBottom: 12, padding: '0 2px' }}>🌿 按功效主治分類</div>
            <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '10px' }}>
              {CATEGORIES.filter(cc => cc.key !== '').map(cc => (
                <button key={cc.key} onClick={() => { setSelectedCat(cc.key); setHerbView('list') }} style={{
                  backgroundColor: '#FFFEF9', borderRadius: 16, padding: '16px 14px',
                  border: '1.5px solid #E8E4DC', cursor: 'pointer', textAlign: 'left',
                  boxShadow: '0 2px 8px rgba(0,0,0,0.06)',
                }}>
                  <div style={{ fontSize: 22, marginBottom: 6 }}>{cc.emoji}</div>
                  <div style={{ fontSize: 14, fontWeight: 800, color: '#1a2C24', marginBottom: 3 }}>{cc.label}</div>
                </button>
              ))}
            </div>
          </div>
        )}

        {herbView === 'list' && (
        <>
        {/* Category pills */}
        <div style={{ padding: '10px 14px 0', overflowX: 'auto', WebkitOverflowScrolling: 'touch' }}>
          <div style={{ display: 'flex', gap: 6, minWidth: 'max-content' }}>
            {CATEGORIES.map(cat => (
              <button key={cat.key} onClick={() => setSelectedCat(cat.key)} style={{
                padding: '6px 12px', borderRadius: 20, border: 'none', cursor: 'pointer', fontSize: 12, fontWeight: 700,
                backgroundColor: selectedCat === cat.key ? '#FFFEF9' : 'rgba(255,254,249,0.15)',
                color: selectedCat === cat.key ? '#1a3A2C' : 'rgba(255,254,249,0.85)',
                whiteSpace: 'nowrap', flexShrink: 0,
              }}>
                {cat.emoji} {cat.label}
              </button>
            ))}
          </div>
        </div>

        {/* Count */}
        <div style={{ padding: '8px 16px 0', fontSize: 12, color: 'rgba(255,254,249,0.65)' }}>
          {loading ? '載入中...' : `${filtered.length} 味中藥`}
        </div>
      </div>

      {/* Content */}
      <div style={{ padding: '12px 14px 100px' }}>
        {loading ? ('''

if old_cat_pills not in c:
    print("HERBS: pattern not found")
    import sys; sys.exit(1)

c = c.replace(old_cat_pills, new_cat_pills)

# Close the herbView === 'list' block after content div
# Find the </div> that closes content div (before bottom sheet)
old_content_end = '''        ) : filtered.length === 0 ? (
          <div style={{ padding: '60px 0', textAlign: 'center', color: '#7A7A6A' }}>
            <div style={{ fontSize: 40, marginBottom: 12 }}>🔍</div>
            <div style={{ fontSize: 15, fontWeight: 600, marginBottom: 4 }}>找不到符合的中藥</div>
            <div style={{ fontSize: 12 }}>試試其他關鍵字或分類</div>
          </div>
        )'''

# The content ends with: ...filtered.map... and then </div> closing the content div
# We need to close the fragment before selectedHerb bottom sheet
# Find: after the last </div> in content, before {selectedHerb && (
# Let's find the closing pattern in original - it should be the </div> closing the flex column
# The original ends with a bunch of </div> and then {selectedHerb &&
# Let me find where selectedHerb bottom sheet starts
# Actually, the content section ends with the closing </div> of the flex column
# Then there's another </div> closing padding:12px
# Then {selectedHerb && (
# We need to close the fragment </> )} after that final </div>

# Find the selectedHerb bottom sheet and insert a closing after the content divs
old_bottom_sheet_start = "      {selectedHerb && ("
idx = c.indexof(old_bottom_sheet_start)
# Insert closing before it
c = c[:idx] + "\n          </div>\n        </>\n        )}\n" + c[idx:]

with open('/home/justin/.openclaw/workspace/tcm_ai/vercel_tcm_frontend/app/[locale]/herbs/page.tsx', 'w') as f:
    f.write(c)
print("herbs done")

File: fix_herbs.js
const fs = require('fs');
let c = fs.readFileSync('/home/justin/.openclaw/workspace/tcm_ai/vercel_tcm_frontend/app/[locale]/herbs/page.tsx', 'utf8');

// 1. Add herbView state
c = c.replace(
  "const [selectedCat, setSelectedCat] = useState('')",
  "const [selectedCat, setSelectedCat] = useState('')\n  const [herbView, setHerbView] = useState<'home' | 'list'>('home')"
);

// 2. Replace the category pills section with a conditional
// Insert herbView === 'home' cards before category pills
const catPillsMarker = "        {/* Category pills */}";
const catPillsIdx = c.indexOf(catPillsMarker);

const homeSection = `        {herbView === 'home' && (
          <div style={{ padding: '16px 14px 100px' }}>
            <div style={{ fontSize: 13, color: 'rgba(255,254,249,0.65)', marginBottom: 12, padding: '0 2px' }}>🌿 按功效主治分類</div>
            <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '10px' }}>
              {CATEGORIES.filter(cc => cc.key !== '').map(cc => (
                <button key={cc.key} onClick={() => { setSelectedCat(cc.key); setHerbView('list') }} style={{
                  backgroundColor: '#FFFEF9', borderRadius: 16, padding: '16px 14px',
                  border: '1.5px solid #E8E4DC', cursor: 'pointer', textAlign: 'left',
                  boxShadow: '0 2px 8px rgba(0,0,0,0.06)',
                }}>
                  <div style={{ fontSize: 22, marginBottom: 6 }}>{cc.emoji}</div>
                  <div style={{ fontSize: 14, fontWeight: 800, color: '#1a2C24', marginBottom: 3 }}>{cc.label}</div>
                </button>
              ))}
            </div>
          </div>
        )}

        {herbView === 'list' && (
        <>
        {/* Category pills */}
`;

const listCloseSection = `
        </>
        )}
`;

c = c.slice(0, catPillsIdx) + homeSection + c.slice(catPillsIdx);

// 3. After the content block, close the herbView === 'list' condition
// Find the last </div> before something that follows the content (like a modal or footer)
// The content div ends, then the herb detail bottom sheet or other UI
// Let me find a good closing point - after the content div's closing tag
const contentDivIdx = c.indexOf("      {/* Content */}");
// Find the </div> that closes the content div - it's the </div> right before the next top-level JSX element
// The structure after content is: {selectedHerb && ( bottom sheet }
// So we need to find the </div> that ends the content div
// Count the divs... Actually let me use a simpler approach - find the closing </div> right after the list/loading block

// After replacement the structure is:
// <div style={{ padding: '12px 14px 100px' }}>  (content)
//   ... list items
//   {loading ? ... : ...}
// </div>
// {selectedHerb && (... bottom sheet ...)}

// So we need to insert "        </>\n        )}\n" after the content </div>
// Let's find the last occurrence of the content closing div pattern
const contentClosing = "<div style={{ padding: '12px 14px 100px' }}>";
// Search for the last one - there should be only one (the one we inserted before)
const lastContentDiv = c.lastIndexOf(contentClosing);
// Find the </div> after it
const afterLastContent = c.slice(lastContentDiv + contentClosing.length);
const closeDivIdx = afterLastContent.indexOf('</div>');
const fullCloseIdx = lastContentDiv + contentClosing.length + closeDivIdx + 6; // +6 for </div>

c = c.slice(0, fullCloseIdx) + '\n          </div>\n        </>\n        )}\n' + c.slice(fullCloseIdx);

fs.writeFileSync('/home/justin/.openclaw/workspace/tcm_ai/vercel_tcm_frontend/app/[locale]/herbs/page.tsx', c);
console.log('done');